examples/sprite_animation/main.rs

```diff
@@ -34,66 +34,99 @@ enum AnimationId {
-                "prefab/sprite_animation.ron",
-                RonFormat,
-                self.progress_counter.as_mut().unwrap(),
-            )
-        });
-        let arrow_test_prefab = world.exec(|loader: PrefabLoader<'_, MyPrefabData>| {
-            loader.load(
-                "prefab/sprite_animation_test.ron",
-                RonFormat,
+
+        {
+            let loader = resources
+                .get_mut::<DefaultLoader>()
+                .expect("Missing loader");
+            // let bat_prefab = loader.load("prefab/sprite_animation.ron");
+            // let arrow_test_prefab = loader.load("prefab/sprite_animation_test.ron");
+
+            let texture = loader.load("texture/bat.32x32.png");
+            let sprites = loader.load("sprites/bat.ron");
+            let sprite_sheet_store = resources.get::<ProcessingQueue<SpriteSheet>>().unwrap();
+            let sheet: Handle<SpriteSheet> =
+                loader.load_from_data(SpriteSheet { texture, sprites }, (), &sprite_sheet_store);
+
+            //let anims: Handle<Sampler<SpriteRenderPrimitive>> = loader.load("anims/bat.ron");
+            let anims = loader.load_from_data(
+                Sampler::<SpriteRenderPrimitive> {
+                    input: vec![0.0, 0.2, 0.4, 0.6, 0.8, 1.0],
+                    output: vec![
+                        SpriteRenderPrimitive::SpriteIndex(4),
+                        SpriteRenderPrimitive::SpriteIndex(3),
+                        SpriteRenderPrimitive::SpriteIndex(2),
+                        SpriteRenderPrimitive::SpriteIndex(1),
+                        SpriteRenderPrimitive::SpriteIndex(0),
+                    ],
+                    function: InterpolationFunction::Step,
+                },
+                (),
+                &resources
+                    .get::<ProcessingQueue<Sampler<SpriteRenderPrimitive>>>()
+                    .unwrap(),
+            );
+            let mut anim_set = AnimationSet::new();
+            let anim_handle: Handle<Animation<SpriteRender>> = loader.load_from_data(
+                Animation::<SpriteRender>::new_single(
+                    0,
+                    amethyst_animation::SpriteRenderChannel::SpriteIndex,
+                    anims,
+                ),
                 self.progress_counter.as_mut().unwrap(),
-            )
-        });
+                &resources
+                    .get::<ProcessingQueue<Animation<SpriteRender>>>()
+                    .unwrap(),
+            );
+            anim_set.insert(AnimationId::Fly, anim_handle);
+
+            world.push((SpriteRender::new(sheet, 0), Transform::default(), anim_set));
+        }
         // Creates new entities with components from MyPrefabData
-        world.create_entity().with(bat_prefab).build();
-        world.create_entity().with(arrow_test_prefab).build();
+        // world.create_entity().with(bat_prefab).build();
+        // world.create_entity().with(arrow_test_prefab).build();
+
```

Comment:
I guess this ok for now. Do we want to revert his when asset loading is fully working?

Reply:
I'm working on prefabs later, plus there is prior conversation throughout the issues of making the examples rely less on prefabs.  The fastest way to get there is to port examples away from prefabs and possibly convert them back later.  We should rely on user feedback as to whether or not they need prefabs for a given example.

This example works in this state to demonstrate an animated sprite so I'm happy with it as is.  I think examples like this one should be focused on demonstrating the programmatic API rather than the magic of prefabs.

I have some work in progress towards a big prefab example that demos a bunch of different prefabs and swapping between them.  With the legion-prefab approach you are essentially just serializing the component directly.